2-WebApp-graph-user/2-1-Call-MSGraph/README.md

```diff
@@ -1,102 +1,222 @@
-1. Find the app key `ClientSecret` and replace the existing value with the key you saved during the creation of the `WebApp-OpenIDConnect-DotNet-code-v2` app, in the Azure portal.
+1. Open the `appsettings.json` file.
+1. Find the key `ClientId` and replace the existing value with the application ID (clientId) of `WebApp-OpenIDConnect-DotNet-graph-v2` app copied from the Azure portal.
+1. Find the key `TenantId` and replace the existing value with your Azure AD tenant ID.
+1. Find the key `Domain` and replace the existing value with your Azure AD tenant name.
+1. Find the key `ClientSecret` and replace the existing value with the key you saved during the creation of `WebApp-OpenIDConnect-DotNet-graph-v2` copied from the Azure portal.
 
 - In case you want to deploy your app in Sovereign or national clouds, ensure the `GraphApiUrl` option matches the one you want. By default this is Microsoft Graph in the Azure public cloud
 
   ```JSon
    "GraphApiUrl": "https://graph.microsoft.com/v1.0"
   ```
 
-### Step 3: Run the sample
+## Running the sample
+
+> For Visual Studio Users
+>
+> Clean the solution, rebuild the solution, and run it.  You might want to go into the solution properties and set the right startup project first.
 
-1. Build the solution and run it.
+```console
+    dotnet run
+```
 
-2. Open your web browser and make a request to the app. The app immediately attempts to authenticate you via the Microsoft identity platform endpoint. Sign in with your personal account or with a work or school account.
+## Explore the sample
 
-3. Go to the **Profile** page, you should now see all kind of information about yourself as well as your picture (a call was made to the Microsoft Graph */me* endpoint)
+1. Open your web browser and make a request to the app at url `https://localhost:44321`. The app immediately attempts to authenticate you via the Microsoft identity platform endpoint. Sign in with a work or school account.
+
+1. Provide consent to the screen presented
+
+1. Click on the **Profile** link on the top menu, you should now see all kind of information about yourself as well as your picture (a call was made to the Microsoft Graph */me* endpoint)
 
 > Did the sample not work for you as expected? Did you encounter issues trying this sample? Then please reach out to us using the [GitHub Issues](../../../../issues) page.
 
 > [Consider taking a moment to share your experience with us.](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRz0h_jLR5HNJlvkZAewyoWxUNEFCQ0FSMFlPQTJURkJZMTRZWVJRNkdRMC4u)
 
 ## About The code
 
-Starting from the [previous phase of the tutorial](../../1-WebApp-OIDC), the code was incrementally updated with the following steps:
+1. In your aspnet core web project, add the `Microsoft.Identity.Web` and `Microsoft.Identity.Web.UI` NuGet packages. It's used to simplify signing-in and acquiring tokens for Microsoft Graph.
```

Comment:
```suggestion
To enable your own ASP.NET Core web application to sign in and call Microsoft Graph like this sample, you must implement the following steps.

### Add the Microsoft.Identity.Web and Microsoft.Identity.Web.UI NuGet packages

In your ASP.NET Core web project, add the `Microsoft.Identity.Web` and `Microsoft.Identity.Web.UI` NuGet packages. These packages are used to simplify signing-in and acquiring tokens for Microsoft Graph.
```

The wording in this entire section needs to be resolved. I think this is supposed to instruct the reader on how to make a new app (rather than follow the sample repo). However this is not communicated to the user at all. Adding some guidance.